### PR TITLE
Update: Deprecate prefer-reflect (fixes #7226)

### LINKED ISSUE #7226: prefer-reflect: Reflect.deleteProperty considered harmful
The rule tells that:
> `Reflect.deleteProperty` effectively deprecates `delete` keyword

Putting conciseness of `delete` and "`Map` should be used instead" points aside, given strict mode enabled and considering only property references, `Reflect.deleteProperty` returns result of `[[Delete]]` call (boolean), but **does not throw** on `false`, unlike `delete` operator does.

Throwing if deletion fails (non-configurable property, `Proxy` trap) is what modern JavaScript does. Not throwing [causes confusion](https://github.com/mgol/check-dependencies/pull/24/files#r76153529).

Quoting @nzakas on `prefer-reflect`:
> This rule should warn only for direct replacements.

I think `Reflect.deleteProperty` should be removed from the rule.

EDIT: same goes for `preventExtensions` and `setPrototypeOf`.

### PR DESCRIPTION
**Pull request by kaicataldo**
**What is the purpose of this pull request? (put an "X" next to item)**

[X] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[X] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

**What changes did you make? (Give an overview)**
Update `prefer-reflect`'s documentation, add the `deprecated` key to `prefer-reflect`'s metadata, and fix our own `conf/eslint.json` to use the replacements for the currently deprecated rules.

**Is there anything you'd like reviewers to focus on?**
Anything I missed?

### CODE REVIEW

File: lib/rules/prefer-reflect.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 /**
  * @fileoverview Rule to suggest using "Reflect" api over Function/Object methods
  * @author Keith Cirkel <http://keithcirkel.co.uk>
+ * @deprecated in ESLint v3.9.0
```
Comment (kaicataldo) [inline lib/rules/prefer-reflect.js]: This assumes this gets in before tomorrow's release - if not, I'll update it.

File: conf/eslint.json [hunk 2]
```diff
@@ -115,7 +115,7 @@
         "no-unneeded-ternary": "off",
         "no-unreachable": "error",
         "no-unsafe-finally": "error",
-        "no-unsafe-negation": "off",
+        "no-unsafe-negation": "error",
```
Comment (vitorbal) [inline conf/eslint.json]: This is a good catch! Should these changes be a separate PR, though, just in case?

Comment (kaicataldo) [inline conf/eslint.json]: I can do that if you'd like :)

**Comment by kaicataldo** [comment]
Ah, thanks for adding the labels - forgot!
